config/karma/shared.karma.conf.js

```diff
@@ -75,13 +75,19 @@ function getConfig(config) {
     coverageIstanbulReporter: {
       combineBrowserReports: true,
       fixWebpackSourcePaths: true,
+      skipFilesWithNoCoverage: true,
```

Comment:
This is the primary change. It's another case of `istanbul` not handling Window's backslashes correctly.

See: https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/src/reporter.js#L105-L117

Reply:
**Some more context:**

The issue is with `karma-coverage-istanbul-reporter`. Setting `skipFilesWithNoCoverage` only works because of a side effect of how it's handling Windows file paths. If you set that property to `false`, it will attempt to locate all files found in the source maps that did not get covered, and add them as "empty" coverage reports. However, it's doing it for _all_ files on Windows, even the ones that do have coverage. This is happening because `istanbul` adds a forward-slash after the path root for all of its remapped file paths. For example, on a Mac the following is generated:

```
/Users/stevebr/Projects/github/skyux-spa-test/src/app/sample/sample.module.ts
```

On Windows, the path looks like (note the `\/src`): 

```
C:\Users\stevebr\Projects\skyux-lib-foobar\/src\app\public\modules\sample\sample.module.ts
```

The `fixPathSeparators` utility "fixes" the forward slash into (note the `\\`):
```
C:\Users\stevebr\Projects\skyux-lib-foobar\\src\app\public\modules\sample\sample.module.ts
```
See: https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/src/util.js#L20

However, the comparison for "empty" coverage files is only looking for _single_ backslashes and adds "empty" files for every single file in the source since the paths now include _double_ backslashes.
See: https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/src/reporter.js#L112

**Things to note:**

- This is only a problem on Windows.
- Setting the `skipFilesWithNoCoverage` to `true` will fix the issue on Windows, and does not change the current behavior for Karma coverage. Currently, we only pull in the source files that are imported into a `*.spec.ts` file, so we never have "empty" files in a coverage report.
- I can't create an issue on `karma-coverage-istanbul-reporter` without being a sponsor on Patreon (super dumb). See: https://github.com/mattlewis92/karma-coverage-istanbul-reporter/issues/new

**Potential workarounds:**
- We merge this as-is, maybe with a descriptive comment above the property.
- We fork `karma-coverage-istanbul-reporter` and make the change ourselves.
- We become sponsors on Patreon.

Reply:
To see it working locally (without setting `skipFilesWithNoCoverage` to `true`, open `./node_modules/karma-coverage-istanbul-reporter/src/util.js` and add the following to line 21:

```
.replace(/\\\\/g, '\\');
```

Line 21, here: https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/src/util.js#L20